Loop/Managers/SleepStore.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  SleepStore.swift
+//  Loop
+//
+//  Created by Anna Quinlan on 12/28/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+
+enum SleepStoreResult<T> {
+    case success(T)
+    case failure(Error)
+}
+
+enum SleepStoreError: Error {
+    case noMatchingBedtime
+    case unknownReturnConfiguration
+    case noSleepDataAvailable
+}
+
+extension SleepStoreError: LocalizedError {
+    public var localizedDescription: String {
+        switch self {
+        case .noMatchingBedtime:
+            return NSLocalizedString("Could not find a matching bedtime", comment: "")
+        case .unknownReturnConfiguration:
+            return NSLocalizedString("Unknown return configuration from query", comment: "")
+        case .noSleepDataAvailable:
+            return NSLocalizedString("No sleep data available", comment: "")
+        }
+    }
+}
+
+class SleepStore {
+    var healthStore: HKHealthStore
+    var sampleLimit: Int
```

Comment:
`sampleLimit` strikes me more as a parameter to the query, rather than as a property of the SleepStore itself. Consider making this a parameter to `getAverageSleepStartTime` with a default value.